### PR TITLE
Adding percent encoding length check for Path.Uri #1553

### PR DESCRIPTION
**Pull request by kencharos**
* Add percent encoding length check at Uri.decode. Bcause
StringIndexOutOfBoundsException raises when Uri.Path call by illegal
format(ex. %AA%, %AA%1).
* Add test case

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @uls-kentaro-maeda,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by ktoso** [comment]
Was nice hakking yesterday, thanks for completing the PR @kencharos !
We'll give it a look with the team and also check the performance impact with benchmarks :)

As the bot above explained we need a confirmation as explained in the above link via the CLA so we can accept a patch (it's basically about you confirming that you're the author of the source and are OK to contribute it).

**Comment by ktoso** [comment]
OK TO TEST

**Comment by kencharos** [comment]
I did  sign the CLA

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

File: akka-http-core/src/test/scala/akka/http/scaladsl/model/UriSpec.scala [hunk 1]
```diff
@@ -597,6 +603,14 @@ class UriSpec extends WordSpec with Matchers {
             "            ^")
       }
 
+      // illgal percent-encoding ends with %
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/UriSpec.scala]: Typo in comment ;)

**Comment by jrudolph** [comment]
Thanks for the quick fix, @ktoso. I keep forgetting that you can now edit right in place in the PR review making these kinds of edits quite convenient.

**Comment by jrudolph** [comment]
Fixes #1553.

**Comment by kencharos** [comment]
@jrudolph @raboof   Thanks for reviewing.
@Koto Thanks for fixing code and advising on last Saturday.
I'm very glad. I'm going to try other issue next time.

**Comment by ktoso** [comment]
> Thanks for the quick fix, @ktoso. I keep forgetting that you can now edit right in place in the PR review making these kinds of edits quite convenient.

Yeah :-) Indeed it's much quicker turnaround if it's just some small fix like here to fix it in-place than do a review cycle.

Thanks a lot @kencharos! ありがとうございました！ ;-)

**Pull request merged**